Sources/mParticle-Appboy/MPKitAppboy.m

```diff
@@ -425,37 +426,105 @@ - (MPKitExecStatus *)routeCommerceEvent:(MPCommerceEvent *)commerceEvent {
+                    [properties addEntriesFromDictionary:productDictionary];
+                }
+                
+                // Strips key/values already being passed to Appboy, plus key/values initialized to default values
+                keys = @[kMPExpProductSKU, kMPProductCurrency, kMPExpProductUnitPrice, kMPExpProductQuantity, kMPProductAffiliation, kMPExpProductCategory, kMPExpProductName];
+                [properties removeObjectsForKeys:keys];
+                
+                [appboyInstance logPurchase:product.sku
+                                   currency:currency
+                                      price:[product.price doubleValue]
+                                   quantity:[product.quantity integerValue]
+                                 properties:properties];
+                
+                [execStatus incrementForwardCount];
+            }
         }
     } else {
-        NSArray *expandedInstructions = [commerceEvent expandedInstructions];
-        
-        for (MPCommerceEventInstruction *commerceEventInstruction in expandedInstructions) {
-            [self logBaseEvent:commerceEventInstruction.event];
+        if (_configuration[bundleProductsWithCommerceEvents] && [_configuration[bundleProductsWithCommerceEvents] boolValue]) {
+            NSDictionary *transformedEventInfo = [commerceEvent.customAttributes transformValuesToString];
+            
+            NSMutableDictionary *eventInfo = [[NSMutableDictionary alloc] initWithCapacity:commerceEvent.customAttributes.count];
+            [transformedEventInfo enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key, id _Nonnull obj, BOOL * _Nonnull stop) {
+                NSString *strippedKey = [self stripCharacter:@"$" fromString:key];
+                eventInfo[strippedKey] = obj;
+            }];
+            
+            if (self->_enableTypeDetection) {
+                eventInfo = [[self simplifiedDictionary:eventInfo] mutableCopy];
+            }
+            
+            NSArray *productArray = [self getProductListParameters:commerceEvent.products];
+            if (productArray.count > 0) {
+                [eventInfo setValue:productArray forKey:productKey];
+            }
+            NSArray *promotionArray = [self getPromotionListParameters:commerceEvent.promotionContainer.promotions];
+            if (promotionArray.count > 0) {
+                [eventInfo setValue:promotionArray forKey:promotionKey];
+            }
+            NSArray *impressionArray = [self getImpressionListParameters:commerceEvent.impressions];
+            if (impressionArray.count > 0) {
+                [eventInfo setValue:impressionArray forKey:impressionKey];
+            }
+            
+            NSString *eventName = [NSString stringWithFormat:@"eCommerce - %@", [self eventNameForAction:commerceEvent.action]];
+            if ([eventName isEqualToString:@"eCommerce - unknown"]) {
+                if (commerceEvent.impressions != nil) {
```

Comment:
```suggestion
                if (commerceEvent.impressions) {
```

Bit of a nitpick, but I notice we generally use this format, so figured we should use it here for consistency.